src/components/legacy/markdown/custom-components.tsx

```diff
@@ -131,19 +149,69 @@ const customSatellytesComponents = {
     return <SubTitle as={'h1'}>{props.children}</SubTitle>;
   },
   h2(props) {
-    return <SubTitle as={'h2'}>{props.children}</SubTitle>;
+    return (
+      <SubTitle as={'h2'}>
+        <a
+          href={'#' + generateAnchorId(props.children)}
+          id={generateAnchorId(props.children)}
+        >
+          #
+        </a>
+        {props.children}
+      </SubTitle>
+    );
   },
   h3(props) {
-    return <TextTitle as={'h3'}>{props.children}</TextTitle>;
+    return (
+      <TextTitle as={'h3'}>
+        <a
+          href={'#' + generateAnchorId(props.children)}
+          id={generateAnchorId(props.children)}
+        >
+          #
+        </a>
+        {props.children}
+      </TextTitle>
+    );
   },
   h4(props) {
-    return <SmallTitle as={'h4'}>{props.children}</SmallTitle>;
+    return (
+      <SmallTitle as={'h4'}>
+        <a
+          href={'#' + generateAnchorId(props.children)}
+          id={generateAnchorId(props.children)}
+        >
```

Comment:
seeing all those anchors nested I think you should not do that. We will not need it everywhere. 
I think it would be better to offer it for some specific cases where we want it (like in our blog or in sections).

Maybe you can create a higher order component and wrap any given header? The anchor doesn't need to stay inside the headline as it's solely an anchor. Something like this:

```
// AnchorizeHOC will prepend an anchor, read the content of the title component to generate the anchor id.
const AnchoredTitle = AnchorizeHOC(TitleComponent)

<AnchoredTitle/>

```

What do you think? Maybe @feedm3 has some thoughts on that ?

Reply:
I converted the current version into a higher-order component and it looks much better now. In addition, it can now be reused more easily, e.g. on the service page (https://satellytescommain21751-featblogpostanchor.gtsb.io/services/#digital-platforms)